aurelia-slickgrid/src/aurelia-slickgrid/filters/compoundInputFilter.ts

```diff
@@ -0,0 +1,203 @@
+import { inject } from 'aurelia-framework';
+import { I18N } from 'aurelia-i18n';
+import { htmlEntityEncode } from './../services/utilities';
+import {
+  Column,
+  FieldType,
+  Filter,
+  FilterArguments,
+  FilterCallback,
+  FilterType,
+  GridOption,
+  OperatorString,
+  OperatorType,
+  SearchTerm
+} from './../models/index';
+
+// using external non-typed js libraries
+declare var $: any;
+
+@inject(I18N)
+export class CompoundInputFilter implements Filter {
+  private $filterElm: any;
+  private $filterInputElm: any;
+  private $selectOperatorElm: any;
+  grid: any;
+  gridOptions: GridOption;
+  operator: OperatorType | OperatorString | undefined;
+  searchTerm: SearchTerm | undefined;
+  columnDef: Column;
+  callback: FilterCallback;
+  filterType = FilterType.compoundInput;
+
+  constructor(private i18n: I18N) { }
+
+  /**
+   * Initialize the Filter
+   */
+  init(args: FilterArguments) {
+    if (args) {
+      this.grid = args.grid;
+      this.callback = args.callback;
+      this.columnDef = args.columnDef;
+      this.operator = args.operator || '';
+      this.searchTerm = args.searchTerm;
+      if (this.grid && typeof this.grid.getOptions === 'function') {
+        this.gridOptions = this.grid.getOptions();
+      }
+
+      // step 1, create the DOM Element of the filter which contain the compound Operator+Input
+      // and initialize it if searchTerm is filled
+      this.$filterElm = this.createDomElement();
+
+      // step 3, subscribe to the keyup event and run the callback when that happens
+      // also add/remove "filled" class for styling purposes
+      this.$filterInputElm.keyup((e: any) => {
+        this.onTriggerEvent(e);
+      });
+      this.$selectOperatorElm.change((e: any) => {
+        this.onTriggerEvent(e);
+      });
+    }
+  }
+
+  /**
+   * Clear the filter value
+   */
+  clear(triggerFilterKeyup = true) {
+    if (this.$filterElm && this.$selectOperatorElm) {
+      this.$selectOperatorElm.val(0);
+      this.$filterInputElm.val('');
+      if (triggerFilterKeyup) {
+        this.$filterElm.trigger('keyup');
+      }
+    }
+  }
+
+  /**
+   * destroy the filter
+   */
+  destroy() {
+    if (this.$filterElm) {
+      this.$filterElm.off('keyup').remove();
```

Comment:
do you need a `this.$selectOperatorElm.off('change')` here?

Reply:
oh right, I could do that one too.